### ISSUE TITLE
Local brokering of Solr Cloud in provided k8s

### ISSUE THREAD

**Issue by mogul** [GSA/data.gov#1943]
## User Story

In order to identify any problems with brokering Solr Cloud quickly, the data.gov team wants to demonstrate OSBAPI brokering of Solr instances into a local user-supplied k8s cluster.

## Acceptance Criteria
GIVEN I have checked out the `datagov-brokerpak` repository \
AND I am running Docker Desktop \
AND I have run `make build up test-env-up wait`
- [ ] WHEN I run `make demo-up` \
  THEN I see that a SolrCloud instance is provisioned
  AND I see a URL for accessing the SolrCloud instance
  AND I can see a Solr dashboard at that URL (...after a little while...)

## Background
This story isolates early validation of brokering Solr Cloud instances *outside* of k8s context, which is an intermediate deliverable for #1808. This avoids the additional complexity of having to broker k8s instances while we validate the core Solr Cloud brokering workflow. (We might also be able to demonstrate this in cloud.gov itself, rather than with Docker Desktop/minikube, but that depends on having a k8s instance up and accessible to cloud.gov applications.)

## Security Considerations ([required](https://nvd.nist.gov/800-53/Rev4/control/CM-4))
[comment]: # "Our SSP says 'The Data.gov team ensures security implications are considered as part of the agile requirements refinement process by including a section in the issue template used as a basis for new work.' so please don't remove this section without care."
This story will result in delivery of the datagov brokerpak into the `datagov-ssb` deployment. All normal configuration management requirements should apply to delivery and deployment of the brokerpak. As the broker will not be useful without supplying k8s details, it's not practically available for usage at the end of this story. Story #1808 should change that.

## Sketch
- Get [this PR merged](https://github.com/GSA/catalog-solr/pull/1), if possible
- Write the Terraform for provisioning a Solr Cloud service using the Helm and Kubernetes providers
- Bundle up the terraform using [brokerpak](https://github.com/GoogleCloudPlatform/gcp-service-broker/blob/master/docs/brokerpak-intro.md)
- Set up continuous delivery of the brokerpak into a GitHub release
- Add the brokerpak during the GitHub Action which runs the datagov-ssb deployment

**Comment by mogul** [comment]
Spent way too much time tonight chasing down, reproducing, and reporting a bug in the k8s Terraform provider.  🙄
https://github.com/hashicorp/terraform-provider-kubernetes/issues/1009

**Comment by mogul** [comment]
I may have run into [a bug with how the `cloud-service-broker` interacts with Terraform 0.13](https://github.com/pivotal/cloud-service-broker/issues/99), which has new registry functionality. I think I can work around this by downgrading Terraform... Will give it a try in the morning.

**Comment by mogul** [comment]
Once the PR referenced above is merged, we have code that meets the Acceptance Criteria above. There are two remaining issues that we'd like to resolve outside the scope of this story:
- We are seeing extraneous computed inputs included in the credentials generated at binding time. We think this is a bug in the `cloud-service-broker` and have [asked the upstream team](https://cloudfoundry.slack.com/archives/C0164KKEZGX/p1604607674096800) about it in Slack.
- We are requiring full k8s credentials of people who just want to broker a SolrCloud instance when really the existence and credentials for finding the k8s should not be their concern. We need those credentials here when we're testing locally with Docker Desktop, but we won't once this is running in an AWS-hosted k8s instance. So we think all those parameters can be made optional. That will be doable once we finish #1808. We'll create a separate issue for that.

**Comment by mogul** [comment]
> * We are seeing extraneous computed inputs included in the credentials generated at binding time. We think this is a bug in the `cloud-service-broker` and have [asked the upstream team](https://cloudfoundry.slack.com/archives/C0164KKEZGX/p1604607674096800) about it in Slack.

We resolved this issue: The fields are still in the output, but they're empty (see the Slack conversation for details as to why).
